keyboards/crkbd/keymaps/cameronjlarsen/keymap.c

```diff
@@ -0,0 +1,363 @@
+        caps_word_on();  // Activate Caps Word!
+      }
+      break;
+
+    // Other combos...
+  }
+}
+
+bool is_oneshot_cancel_key(uint16_t keycode){
+  switch (keycode) {
+    case LA_NAV:
+      return true;
+    default:
+      return false;
+  }
+}
+
+bool is_oneshot_ignored_key(uint16_t keycode){
+  switch (keycode) {
+    case LA_NAV:
+    case LA_SYM:
+    case OS_SHFT:
+    case OS_CTRL:
+    case OS_ALT:
+    case OS_GUI:
+      return true;
+    default:
+      return false;
+  }
+}
+
+oneshot_state os_shft_state = os_up_unqueued;
+oneshot_state os_ctrl_state = os_up_unqueued;
+oneshot_state os_alt_state = os_up_unqueued;
+oneshot_state os_cmd_state = os_up_unqueued;
+
+
+
+bool caps_word_press_user(uint16_t keycode) {
+  switch (keycode) {
+    // Keycodes that continue Caps Word, with shift applied.
+    case KC_A ... KC_Z:
+    case KC_MINS:
+      add_weak_mods(MOD_BIT(KC_LSFT));  // Apply shift to the next key.
+      return true;
+
+    // Keycodes that continue Caps Word, without shifting.
+    case KC_1 ... KC_0:
+    case KC_BSPC:
+    case KC_DEL:
+    case KC_UNDS:
+      return true;
+
+    default:
+      return false;  // Deactivate Caps Word.
+  }
+}
+
+void matrix_scan_user(void) {
+  // Other tasks...
+}
+
+layer_state_t layer_state_set_user(layer_state_t state){
+  return update_tri_layer_state(state, SYM, NAV, FUN);
+}
+
+#ifdef OLED_ENABLE
+oled_rotation_t oled_init_user(oled_rotation_t rotation) {
+  if (!is_keyboard_master()) {
+    return OLED_ROTATION_180;  // flips the display 180 degrees if offhand
+  }
+  return rotation;
+}
+
+void oled_render_layer_state(void) {
+    oled_write_P(PSTR("Layer: "), false);
+    switch (get_highest_layer(layer_state | default_layer_state)) {
+        case _QWERTY:
+            oled_write_ln_P(PSTR("QWERTY"), false);
+            break;
+        case SYM:
+            oled_write_ln_P(PSTR("Sym"), false);
+            break;
+        case NAV:
+            oled_write_ln_P(PSTR("Nav"), false);
+            break;
+        case FUN:
+            oled_write_ln_P(PSTR("Function"), false);
+            break;
+        default:
+            oled_write_ln_P(PSTR("Undefined"), false);
+    }
+}
+
+char keylog_str[24] = {};
+
+const char code_to_name[60] = {
+    ' ', ' ', ' ', ' ', 'a', 'b', 'c', 'd', 'e', 'f',
+    'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p',
+    'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z',
+    '1', '2', '3', '4', '5', '6', '7', '8', '9', '0',
+    'R', 'E', 'B', 'T', '_', '-', '=', '[', ']', '\\',
+    '#', ';', '\'', '`', ',', '.', '/', ' ', ' ', ' '};
+
+void set_keylog(uint16_t keycode, keyrecord_t *record) {
+  char name = ' ';
+    if ((keycode >= QK_MOD_TAP && keycode <= QK_MOD_TAP_MAX) ||
+        (keycode >= QK_LAYER_TAP && keycode <= QK_LAYER_TAP_MAX)) { keycode = keycode & 0xFF; }
+  if (keycode < 60) {
+    name = code_to_name[keycode];
+  }
+
+  // update keylog
+  snprintf(keylog_str, sizeof(keylog_str), "%dx%d, k%2d : %c",
```

Comment:
snprintf is expensive, in terms of firmware space, and can use ~1.5kb of firmware space. 

It may be worth converting this block to not use it.

Reply:
This was from the default crkbd keymap. I'll look into moving away from it. Thanks for the tip.